// Proto format used by the open source incxfer analysis code.
syntax = "proto3";

package analysis;

message PageContentProto {
  string font_name = 1;
  repeated uint32 codepoints = 2;
  repeated uint32 glyph_ids = 4;

  // This is used to add data from logs of existing PFE method(s).
  // Records the resulting request/response sizes incurred by that
  // method for this page view.
  //
  // The analysis will generate costs for this method along with
  // the simulated methods.
  repeated PfeRequest logged_requests = 3;
}

message PfeRequest {
  uint32 request_size = 1;
  uint32 response_size = 2;
}

message PageViewProto {
  repeated PageContentProto contents = 1;
  reserved 2;  // Was language
}

message PageViewSequenceProto {
  repeated PageViewProto page_views = 1;
  // BCP47 Language tag, for which language the majority of the page views are.
  string language = 2;
}

message DataSetProto {
  repeated PageViewSequenceProto sequences = 1;

  // If logged requests are included this is the name
  // of the PFE_method they came from.
  string logged_method_name = 2;
}
